serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -107,24 +114,27 @@ def add(self, request_states: list[RequestState]):
             # States which have been invalidated should never be added, directly
             # cancel them instead.
             valid_states = []
-            for request_state in request_states:
+            kv_cache_size = self.cache_manager.get_kv_cache_size()
+            max_prompt_len = min(self.max_context_length, self.max_num_batched_tokens)
```

Comment:
I don't see why it is a good idea to allow users to run a model on input exceeding the max context length. Technically it is possible, but I think it's better not to do it.

Reply:
Yes, I agree with you. I just thought maybe it would be better to put such constraints at ollm level since it is not the limitation of the engine. It is more like a service-level decision. ollm already has such checks, so the engine wouldn't see the request above `max_sequence_length` in practice. https://github.com/octoml/ollm/blob/main/ollm/engines/mlc_serve.py#L225

Reply:
just found that the vllm also uses this condition https://github.com/vllm-project/vllm/blob/main/vllm/core/scheduler.py#L66-L67